tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -571,127 +504,127 @@ def tridiag_forward(
-        if left < right
-        else (np.asarray(b_I_float), np.asarray(a_I_float))
+    cases.verify(
+        cartesian_case, testee, a, b, left, right, out=out, ref=((a, b) if left < right else (b, a))
     )
-    assert np.allclose(e, out_I_float)
-    assert np.allclose(f, out_I_float_1)
 
 
-def test_ternary_builtin_neighbor_sum(reduction_setup, fieldview_backend):
-    rs = reduction_setup
-    V2EDim = rs.V2EDim
-    V2E = rs.V2E
-
-    num_vertices = rs.num_vertices
-    num_edges = rs.num_edges
-
-    a = gtx.np_as_located_field(Edge)(np.ones((num_edges,)))
-    b = gtx.np_as_located_field(Edge)(2 * np.ones((num_edges,)))
-    out = gtx.np_as_located_field(Vertex)(np.zeros((num_vertices,)))
+def test_ternary_builtin_neighbor_sum(unstructured_case):
+    @gtx.field_operator
+    def testee(a: cases.EField, b: cases.EField) -> cases.VField:
+        tmp = neighbor_sum(b(V2E) if 2 < 3 else a(V2E), axis=V2EDim)
+        return tmp
 
-    @gtx.field_operator(backend=fieldview_backend)
-    def ternary_reduce(
-        a: gtx.Field[[Edge], float], b: gtx.Field[[Edge], float]
-    ) -> gtx.Field[[Vertex], float]:
-        out = neighbor_sum(b(V2E) if 2 < 3 else a(V2E), axis=V2EDim)
-        return out
-
-    ternary_reduce(a, b, out=out, offset_provider=rs.offset_provider)
-
-    expected = (
-        np.sum(np.asarray(b)[rs.v2e_table], axis=1)
-        if 2 < 3
-        else np.sum(np.asarray(a)[rs.v2e_table], axis=1)
+    cases.verify_with_default_data(
+        unstructured_case,
+        testee,
+        ref=lambda a, b: (
+            np.sum(b[unstructured_case.offset_provider["V2E"].table[:, :]], axis=1)
+            if 2 < 3
+            else np.sum(a[unstructured_case.offset_provider["V2E"].table[:, :]], axis=1)
+        ),
     )
 
-    assert np.allclose(expected, out)
-
 
-def test_ternary_scan(fieldview_backend):
-    init = 0.0
-    a_float = 4
-    a = gtx.np_as_located_field(KDim)(a_float * np.ones((size,)))
-    out = gtx.np_as_located_field(KDim)(np.zeros((size,)))
-    expected = np.asarray([i if i <= a_float else a_float + 1 for i in range(1, size + 1)])
-
-    @gtx.scan_operator(axis=KDim, forward=True, init=init, backend=fieldview_backend)
+def test_ternary_scan(cartesian_case):
+    @gtx.scan_operator(axis=KDim, forward=True, init=0.0)
     def simple_scan_operator(carry: float, a: float) -> float:
         return carry if carry > a else carry + 1.0
 
-    simple_scan_operator(a, out=out, offset_provider={})
+    k_size = cartesian_case.default_sizes[KDim]
+    a = gtx.np_as_located_field(KDim)(4.0 * np.ones((k_size,)))
+    out = gtx.np_as_located_field(KDim)(np.zeros((k_size,)))
 
-    assert np.allclose(expected, out)
+    cases.verify(
+        cartesian_case,
+        simple_scan_operator,
+        a,
+        out=out,
+        ref=np.asarray([i if i <= 4.0 else 4.0 + 1 for i in range(1, k_size + 1)]),
+    )
 
 
 @pytest.mark.parametrize("forward", [True, False])
```

Comment:
By introducing a `program` calling the `scan_operator` we can even use default data.

```python
@pytest.mark.parametrize("forward", [True, False])
def test_scan_nested_tuple_output(forward, cartesian_case):
    init = (1, (2, 3))
    k_size = cartesian_case.default_sizes[KDim]
    expected = np.arange(1, 1 + k_size, 1, dtype=int64)
    if not forward:
        expected = np.flip(expected)

    @gtx.scan_operator(axis=KDim, forward=forward, init=init)
    def simple_scan_operator(carry: tuple[int, tuple[int, int]]) -> tuple[int, tuple[int, int]]:
        return (carry[0] + 1, (carry[1][0] + 1, carry[1][1] + 1))

    @gtx.program
    def testee(out: tuple[cases.KField, tuple[cases.KField, cases.KField]]):
        simple_scan_operator(out=out)

    cases.verify_with_default_data(
        cartesian_case,
        testee,
        #  out=(out1, (out2, out3)),
        ref=lambda: (expected + 1.0, (expected + 2.0, expected + 3.0)),
        comparison=lambda ref, out: np.all(out[0].array() == ref[0])
        and np.all(out[1][0].array() == ref[1][0])
        and np.all(out[1][1].array() == ref[1][1]),
    )
```